### PR TITLE
Notify about performance degradation in dataset batch

### PR DESCRIPTION
**Pull request by fridex**
## Related issues or additional information of the supplied change

Related: https://github.com/tensorflow/tensorflow/issues/41827

### CODE REVIEW

**Review by harshad16 — COMMENTED** [review]
/lgtm

as `tensorflow`  package itself has the gpu suport in verson 2.x 
https://pypi.org/project/tensorflow/
how would one state these features?

**Comment by fridex** [comment]
> /lgtm
> 
> as `tensorflow` package itself has the gpu suport in verson 2.x

Good point, adjusted in 39c0f2c.

> how would one state these features?

I think we could extend the YAML configuration so that it can match multiple package versions. I see this "copying units" pattern repeating also for other cases - e.g. tensorflow-cpu/tensorflow-gpu/intel-tensorflow.

File: prescription.yaml [hunk 1]
```diff
@@ -1344,3 +1344,37 @@ spec:
...
+              not: [null]
+        run:
+          match:
+            state:
+              resolved_dependencies:
+                - name: tensorflow
+                  version: ">=2.3.0"
+          justification:
+            - type: INFO
+              message: "pass drop_remainder=True to dataset batch when running tensorflow-gpu>=2.3.0 to improve training performance by 15%"
```
Comment (harshad16) [inline prescription.yaml:1362]:
```suggestion
              message: "pass drop_remainder=True to dataset batch when running tensorflow>=2.3.0 to improve training performance by 15%"
```

**Review by harshad16 — COMMENTED** [review]
/lgtm
one minor suggestion

**Review by harshad16 — APPROVED** [review]
/lgtm

**Pull request merged**